src/sources/nginx/mod.rs

```diff
@@ -0,0 +1,283 @@
+        shutdown: ShutdownSignal,
+        out: Pipeline,
+    ) -> crate::Result<super::Source> {
+        let tls = TlsSettings::from_options(&self.tls)?;
+        let http_client = HttpClient::new(tls)?;
+
+        let namespace = Some(self.namespace.clone()).filter(|namespace| !namespace.is_empty());
+        let sources: Vec<_> = self
+            .endpoints
+            .iter()
+            .map(|endpoint| NginxMetrics {
+                http_client: http_client.clone(),
+                endpoint: endpoint.clone(),
+                auth: self.auth.clone(),
+                namespace: namespace.clone(),
+            })
+            .collect();
+
+        let mut out = out
+            .sink_map_err(|error| error!(message = "Error sending mongodb metrics.", %error))
+            .sink_compat();
+
+        let duration = time::Duration::from_secs(self.scrape_interval_secs);
+        Ok(Box::pin(async move {
+            let mut interval = time::interval(duration).take_until(shutdown);
+            while interval.next().await.is_some() {
+                let start = Instant::now();
+                let metrics = join_all(sources.iter().map(|nginx| nginx.collect())).await;
+                emit!(NginxMetricsCollectCompleted {
+                    start,
+                    end: Instant::now()
+                });
+
+                let mut stream = stream::iter(metrics).flatten().map(Event::Metric).map(Ok);
+                out.send_all(&mut stream).await?;
+            }
+
+            Ok(())
+        }))
+    }
+
+    fn output_type(&self) -> DataType {
+        DataType::Metric
+    }
+
+    fn source_type(&self) -> &'static str {
+        "nginx_metrics"
+    }
+}
+
+#[derive(Debug)]
+struct NginxMetrics {
+    http_client: HttpClient,
+    endpoint: String,
+    auth: Option<Auth>,
+    namespace: Option<String>,
+}
+
+impl NginxMetrics {
+    async fn collect(&self) -> stream::BoxStream<'static, Metric> {
+        let (up_value, metrics) = match self.collect_metrics().await {
+            Ok(metrics) => (1.0, metrics),
+            Err(()) => (0.0, vec![]),
+        };
+
+        stream::once(ready(self.create_metric("up", gauge!(up_value))))
+            .chain(stream::iter(metrics))
+            .boxed()
+    }
+
+    async fn collect_metrics(&self) -> Result<Vec<Metric>, ()> {
+        let response = self.get_nginx_response().await.map_err(|error| {
+            emit!(NginxMetricsRequestError {
+                error,
+                endpoint: &self.endpoint,
+            })
+        })?;
+
+        let status = match String::from_utf8_lossy(&response) {
+            Cow::Borrowed(data) => NginxStubStatus::try_from(data),
+            Cow::Owned(data) => NginxStubStatus::try_from(data.as_str()),
+        }
+        .map_err(|error| {
+            emit!(NginxMetricsStubStatusParseError {
+                error,
+                endpoint: &self.endpoint,
+            })
+        })?;
+
+        Ok(vec![
```

Comment:
Per the RFC, I think we need to be tagging these with `host` and `endpoint` so that it is possible to distinguish them.

Reply:
We should also document those tags in the cue docs.